client/extensions/woocommerce/woocommerce-services/state/shipping-label/test/selectors.js

```diff
@@ -580,3 +580,62 @@ describe( 'Shipping label selectors', () => {
 		} );
 	} );
 } );
+
+describe( 'Shipping label form validation', () => {
+	const siteId = 1;
+	const addressData = {
+		values: {
+			company: 'Automaggic',
+			address_2: '',
+			city: 'Cupertino',
+			state: 'CA',
+			postcode: '95014-0642',
+			country: 'US',
+			phone: '',
+			name: 'Peter Anderson',
+			address: 'Apple Park Way1',
+		}
+	};
+	const appState = {};
+
+	it ( 'with no destination phone entered and destination phone required, return error', () => {
+		const fieldsToValidate = { destinationPhone: true };
+		const errors = getRawAddressErrors( appState, addressData, siteId, fieldsToValidate );
+		expect( errors ).to.be.an( 'Object' ).and.to.have.property( 'phone' );
+	} );
+
+	it ( 'with no destination phone entered and destination phone not required, do not return error', () => {
+		const fieldsToValidate = {};
+		const errors = getRawAddressErrors( appState, addressData, siteId, fieldsToValidate );
+		expect( errors ).to.be.an( 'Object' ).and.to.not.have.property( 'phone' );
+	} );
+
+	it ( 'with no origin phone entered and origin phone required, return error', () => {
+		const fieldsToValidate = { originPhone: true };
+		const errors = getRawAddressErrors( appState, addressData, siteId, fieldsToValidate );
+		expect( errors ).to.be.an( 'Object' ).and.to.have.property( 'phone' );
+	} );
+
+	it ( 'with no origin phone entered and origin phone not required, do not return error', () => {
```

Comment:
Is this the same test cases as line 608?

Reply:
Oh i see why the codes are the same. Because the `addressData` could be from origin or destination, so the `phone: ''` could be empty and `fieldsToValidate = {};`.

Reply:
Yeah, they are technically the same but I thought I would make each case combination explicit.